Sming/Arch/Esp8266/Components/esp8266/startup.cpp

```diff
@@ -134,3 +134,11 @@ namespace std {
         while(1);
     };
 }
+
+extern "C" {
+    __attribute__((weak)) int *__errno(void) 
```

Comment:
Don't need this, add `#include <sys/errno.h>` somewhere.
(See `Arch/Esp8266/Components/gdbstub/gdbstub.cpp`.)

Reply:
This header is already included by <errno.h>. I don't see how including it explicitly should fix the linker error. At least in my toolchain built, sys/errno.h contains the following:

```
#define errno (*__errno())
extern int *__errno _PARAMS ((void));
```

I.e. to use errno, a definition of the `__errno` function has to be provided somewhere. Naturally, it should be in libc, but since Sming uses a stripped-down version of libc, it is not there, which is why I added it to startup.cpp (seemed the logical place, since it is a similar situation with the missing `__throw_bad_function_call`).

Please advise.

Reply:
Strange, it seems `errno` isn't actually used anywhere else (just the codes).
`libmicroc` has a definition for `__errno_location` which should probably be referenced in a header somewhere, but it isn't.

https://github.com/anakod/esp_microc/blob/master/errno/__errno_location.c

I'll raise another PR for this.

Reply:
Ok, I'll rebase again when https://github.com/SmingHub/Sming/pull/1801 is merged and then remove the definition in startup.cpp

Reply:
> Ok, I'll rebase again when #1801 is merged

PR #1801 is merged in develop.